src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInstance.cs

```diff
@@ -4,18 +4,17 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.Azure.WebJobs.Host.Protocols;
-using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Azure.WebJobs.Host.Executors
 {
     internal class FunctionInstance : IFunctionInstanceEx, IDisposable
     {
-        private IServiceScope _instanceServicesScope;
-        private readonly IServiceScopeFactory _serviceScopeFactory;
-        private IServiceProvider _instanceServices;
+        
+        private readonly IInstanceServicesFeatureFactory _instanceServicesFeatureFactory;
+        private IInstanceServicesFeature _instanceServicesFeature;
 
         public FunctionInstance(Guid id, IDictionary<string, string> triggerDetails, Guid? parentId, ExecutionReason reason, IBindingSource bindingSource,
-            IFunctionInvoker invoker, FunctionDescriptor functionDescriptor, IServiceScopeFactory serviceScopeFactory)
+            IFunctionInvoker invoker, FunctionDescriptor functionDescriptor, IInstanceServicesFeatureFactory instanceServiceFeatureFactory)
```

Comment:
I think the whole "Features" concept and naming is an Http specific thing in ASP.NET Core and might not make conceptual sense here. We don't use "Features" anywhere else in this SDK.

Instead, why isn't this just an IFunctionInstanceServiceProviderFactory interface or something?

Reply:
There isn't anything HTTP specific about this. 

Though we're not fully implementing the pattern here, this aligns with it and is also something we use in the worker (https://github.com/Azure/azure-functions-dotnet-worker/blob/6647c454b1820711125a2639b0a3fee47dffc088/src/DotNetWorker.Core/Context/FunctionContext.cs#L62)

As discussed, this still requires both interfaces, but I'm changing the name here.